### ISSUE TITLE
Problem with merges if some replica stopped

### ISSUE THREAD

**Issue by wawanawna** [ClickHouse/ClickHouse#2755]
We have a CH cluster (v 1.1.54385), 8 nodes, 4 shard * 2 replicas.

Writing rate (we insert into MetricsDistributed table, see below) is something like 30k-50k rows in each batch, ~100 batches per minute.

We have Metrics table on each node
```
CREATE TABLE IF NOT EXISTS default.Metrics ON CLUSTER metrics
(
    appId UInt32,
    metricId UInt32,
    timestamp UInt32,
    value Float64,
    `tags.name` Array(UInt32),
    `tags.value` Array(UInt32)
)
ENGINE = ReplicatedMergeTree('/clickhouse/tables/metrics/{shard}/Metrics', '{replica}')
PARTITION BY (floor(timestamp/86400)) ORDER BY (appId, metricId, timestamp);
```

Distributed table on each node
```
CREATE TABLE IF NOT EXISTS  default.MetricsDistributed ON CLUSTER metrics AS default.Metrics
ENGINE = Distributed(metrics, default, Metrics, rand());

```

And MV with its own Distributed table on each node

```
CREATE MATERIALIZED VIEW default.MetricsSampledView  ON CLUSTER metrics
ENGINE = ReplicatedMergeTree('/clickhouse/tables/metrics/{shard}/MetricsSampledView', '{replica}')
PARTITION BY floor(timestamp / 86400)
ORDER BY (appId, metricId, timestamp) POPULATE AS
SELECT *
FROM default.Metrics
WHERE intHash32(timestamp) % 20 = 0;
```

```
CREATE TABLE IF NOT EXISTS  default.MetricsDistributedSampledView ON CLUSTER metrics AS default.MetricsSampledView
ENGINE = Distributed(metrics, default, MetricsSampledView, rand());
```

In front of CH cluster there is load balancer, queries execute on Distributed tables.

Before statring use CH on prod we decided to perfor some HA test. We just stopped CH on one node (with leaders).

Problems:

1. With time number of parts started growing here ```SELECT count(*),table FROM system.parts WHERE active group by table;``` (we checked on other replica)

Then we started getting error on client side 
```
2018.07.27 10:26:01.949040 [ 638 ] <Error> executeQuery: Code: 252, e.displayText() = DB::Exception: Too many parts (302). Merges are processing significantly slower than inserts., e.what() = DB::Exception (from 10.1.2.5:47558) (in query: INSERT INTO MetricsDistributed  FORMAT TabSeparated), Stack trace:
2018.07.27 10:26:02.010363 [ 638 ] <Error> HTTPHandler: Code: 252, e.displayText() = DB::Exception: Too many parts (302). Merges are processing significantly slower than inserts., 
```

What we see in system.replicas (on other node) is that leader was reassigned, this is ok, but errors in tjer table are (part of errors):

```

┌─table─────────────────────┬─postpone_reason─┬─last_exception───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ .inner.MetricsSampledView │                 │                                                                                                                                                                  │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1875_1875_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1904_1904_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │                                                                                                                                                                  │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1869_1869_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1897_1897_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1867_1867_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1940_1940_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │                                                                                                                                                                  │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1881_1881_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │                                                                                                                                                                  │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1919_1919_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1928_1928_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │                                                                                                                                                                  │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1868_1868_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │                                                                                                                                                                  │
│ .inner.MetricsSampledView │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_1934_1934_0 or covering part, e.what() = DB::Exception   │
│ .inner.MetricsSampledView │                 │                                                                                                                                                                  │

                                                                                                                       │
│ Metrics                   │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_42206_42206_0 or covering part, e.what() = DB::Exception │
│ Metrics                   │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_42289_42289_0 or covering part, e.what() = DB::Exception │
│ Metrics                   │                 │                                                                                                                                                                  │
│ Metrics                   │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_42243_42243_0 or covering part, e.what() = DB::Exception │
│ Metrics                   │                 │                                                                                                                                                                  │
│ Metrics                   │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_42212_42212_0 or covering part, e.what() = DB::Exception │
│ Metrics                   │                 │                                                                                                                                                                  │
│ Metrics                   │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_42257_42257_0 or covering part, e.what() = DB::Exception │
│ Metrics                   │                 │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_42228_42228_0 or covering part, e.what() = DB::Exception │

│ .inner.MetricsSampledView │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_2877_2942_14 because part 722f7ee72516522ffdfc83494465fd4e_2877_2938_13 is not ready yet (log entry for that part is being processed).    │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                  │
│ .inner.MetricsSampledView │ Not executing log entry for part 722f7ee72516522ffdfc83494465fd4e_2877_2954_17 because another log entry for the same part is being processed. This shouldn\'t happen often.                     │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                  │
│ .inner.MetricsSampledView │ Not executing log entry for part 722f7ee72516522ffdfc83494465fd4e_2877_2964_20 because another log entry for the same part is being processed. This shouldn\'t happen often.                     │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                  │
│ .inner.MetricsSampledView │ Not executing log entry for part 722f7ee72516522ffdfc83494465fd4e_2877_2970_22 because another log entry for the same part is being processed. This shouldn\'t happen often.

```

2. This is not all, main problem was that after our batcher tried to repeat insert few times we started getting duplications in table (max_insert_block_size is default and our batch size significantly lower than this value, so can't be a problem).

If we DROP MV and repeat test number of parts will grow too and we will see the same errors in log and in table, but with time numbers will stabilize and clsuter will continue working without ```Too many parts (302). Merges are processing significantly slower than inserts.``` issue and without duplications in main table. 

Could you please provide a receipt how to avoid such problems, if you need more info we always ready to provide.

PS: Alex Zatelepin @ztlpn  can provide more info, I already contact him, big thx Alex.

PSS: thx for a great product

**Comment by wawanawna** [comment]
UPDATE: after few tries was able to reproduce problems with merges even without MV. 
After stop 1 replica (node21 with leaders) on node11 (replica) we started seen
```
2018.07.30 13:23:39.715741 [ 35 ] <Error> executeQuery: Code: 252, e.displayText() = DB::Exception: Too many parts (300). Merges are processing significantly slower than inserts., e.what() = DB::Exception (from 10.1.2.97:7098) (in query: INSERT INTO default.MetricsFree FORMAT TabSeparated), Stack trace:

0. /usr/bin/clickhouse-server(StackTrace::StackTrace()+0x15) [0x84eb125]
1. /usr/bin/clickhouse-server(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x21) [0x2ccbf01]
2. /usr/bin/clickhouse-server(DB::MergeTreeData::delayInsertIfNeeded(Poco::Event*)+0x3dc) [0x7e978ec]
3. /usr/bin/clickhouse-server(DB::ReplicatedMergeTreeBlockOutputStream::write(DB::Block const&)+0x4e) [0x7f18bde]
4. /usr/bin/clickhouse-server(DB::PushingToViewsBlockOutputStream::write(DB::Block const&)+0x42) [0x8038e22]
5. /usr/bin/clickhouse-server(DB::AddingDefaultBlockOutputStream::write(DB::Block const&)+0x6b7) [0x7f99f77]
6. /usr/bin/clickhouse-server(DB::SquashingBlockOutputStream::finalize()+0x247) [0x803e767]
7. /usr/bin/clickhouse-server(DB::SquashingBlockOutputStream::writeSuffix()+0x10) [0x803e9f0]
8. /usr/bin/clickhouse-server(DB::TCPHandler::processInsertQuery(DB::Settings const&)+0x331) [0x2cdc111]
9. /usr/bin/clickhouse-server(DB::TCPHandler::runImpl()+0x476) [0x2cdc666]
10. /usr/bin/clickhouse-server(DB::TCPHandler::run()+0x2a) [0x2cdd4ba]
11. /usr/bin/clickhouse-server(Poco::Net::TCPServerConnection::start()+0xe) [0x86d549e]
12. /usr/bin/clickhouse-server(Poco::Net::TCPServerDispatcher::run()+0x169) [0x86d5879]
13. /usr/bin/clickhouse-server(Poco::PooledThread::run()+0x76) [0x8776a36]
14. /usr/bin/clickhouse-server(Poco::ThreadImpl::runnableEntry(void*)+0x37) [0x8772c47]
15. /usr/bin/clickhouse-server() [0x8dd3a8e]
16. /lib/x86_64-linux-gnu/libpthread.so.0(+0x76b9) [0x7f1843a106b9]
17. /lib/x86_64-linux-gnu/libc.so.6(clone+0x6c) [0x7f184323941c]
```

Parts on nodes showed us
```
SELECT table,count(*) FROM system.parts WHERE active group by table;

node11
MetricsFree	740
Metrics	354
node12
MetricsFree	19
Metrics	235
node13
MetricsFree	17
Metrics	241
node14
MetricsFree	14
Metrics	252
node21
node22
MetricsFree	19
Metrics	236
node23
MetricsFree	19
Metrics	244
node24
MetricsFree	18
Metrics	255
```

So obviously merges stopped on node11 because is leader were stopped (node21)

Table *system.replication_queue* had at this time:

```
SELECT 
    table, 
    postpone_reason, 
    last_exception
FROM system.replication_queue 

┌─table───────┬─postpone_reason─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┬─last_exception───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57302_57310_2 because part 722f7ee72516522ffdfc83494465fd4e_57308_57308_0 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57254_57301_8 because part 722f7ee72516522ffdfc83494465fd4e_57254_57292_7 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57302_57325_5 because part 722f7ee72516522ffdfc83494465fd4e_57302_57320_4 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │                                                                                                                                                                                                 │                                                                                                                                                                  │
│ Metrics     │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57302_57315_3 because part 722f7ee72516522ffdfc83494465fd4e_57302_57310_2 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │                                                                                                                                                                                                 │                                                                                                                                                                  │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57254_57330_9 because part 722f7ee72516522ffdfc83494465fd4e_57328_57328_0 is not ready yet (log entry for that part is being processed). │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_57254_57330_9 or covering part, e.what() = DB::Exception │
│ Metrics     │                                                                                                                                                                                                 │                                                                                                                                                                  │
│ Metrics     │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57331_57340_2 because part 722f7ee72516522ffdfc83494465fd4e_57331_57336_1 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57302_57320_4 because part 722f7ee72516522ffdfc83494465fd4e_57302_57315_3 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57331_57344_3 because part 722f7ee72516522ffdfc83494465fd4e_57331_57340_2 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57254_57292_7 because part 722f7ee72516522ffdfc83494465fd4e_57291_57291_0 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │                                                                                                                                                                                                 │                                                                                                                                                                  │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57331_57349_4 because part 722f7ee72516522ffdfc83494465fd4e_57331_57344_3 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ Metrics     │ Not merging into part 722f7ee72516522ffdfc83494465fd4e_57331_57369_8 because part 722f7ee72516522ffdfc83494465fd4e_57331_57364_7 is not ready yet (log entry for that part is being processed). │ Code: 234, e.displayText() = DB::Exception: No active replica has part 722f7ee72516522ffdfc83494465fd4e_57331_57369_8 or covering part, e.what() = DB::Exception │
│ MetricsFree │ Not merging into part 561fcdf2f7bd21b8eb12a475d96caf5d_519_551_5 because part 561fcdf2f7bd21b8eb12a475d96caf5d_550_550_0 is not ready yet (log entry for that part is being processed).         │ Code: 234, e.displayText() = DB::Exception: No active replica has part 561fcdf2f7bd21b8eb12a475d96caf5d_519_551_5 or covering part, e.what() = DB::Exception     │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1256_154 because part a997169d040e813fc6f3049a156c1fbc_1009_1255_153 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1264_162 because part a997169d040e813fc6f3049a156c1fbc_1009_1263_161 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1266_164 because part a997169d040e813fc6f3049a156c1fbc_1009_1265_163 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1269_167 because part a997169d040e813fc6f3049a156c1fbc_1009_1268_166 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1257_155 because part a997169d040e813fc6f3049a156c1fbc_1009_1256_154 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1265_163 because part a997169d040e813fc6f3049a156c1fbc_1009_1264_162 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1250_1255_1 because part a997169d040e813fc6f3049a156c1fbc_1250_1250_0 is not ready yet (log entry for that part is being processed).     │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1258_156 because part a997169d040e813fc6f3049a156c1fbc_1009_1257_155 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1267_165 because part a997169d040e813fc6f3049a156c1fbc_1009_1266_164 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1271_169 because part a997169d040e813fc6f3049a156c1fbc_1009_1270_168 is not ready yet (log entry for that part is being processed). │                                                                                                                                                                  │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1274_172 because part a997169d040e813fc6f3049a156c1fbc_1009_1273_171 is not ready yet (log entry for that part is being processed). │                                                                                                                                                                  │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1276_174 because part a997169d040e813fc6f3049a156c1fbc_1009_1275_173 is not ready yet (log entry for that part is being processed). │                                                                                                                                                                  │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │                                                                                                                                                                  │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │                                                                                                                                                                  │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1263_161 because part a997169d040e813fc6f3049a156c1fbc_1009_1262_160 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1268_166 because part a997169d040e813fc6f3049a156c1fbc_1009_1267_165 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1270_168 because part a997169d040e813fc6f3049a156c1fbc_1009_1269_167 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1272_170 because part a997169d040e813fc6f3049a156c1fbc_1009_1271_169 is not ready yet (log entry for that part is being processed). │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1275_173 because part a997169d040e813fc6f3049a156c1fbc_1009_1274_172 is not ready yet (log entry for that part is being processed). │                                                                                                                                                                  │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1278_176 because part a997169d040e813fc6f3049a156c1fbc_1009_1277_175 is not ready yet (log entry for that part is being processed). │                                                                                                                                                                  │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │                                                                                                                                                                                                 │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part 561fcdf2f7bd21b8eb12a475d96caf5d_552_567_3 because part 561fcdf2f7bd21b8eb12a475d96caf5d_552_562_2 is not ready yet (log entry for that part is being processed).         │ Poco::Exception. Code: 1000, e.code() = 111, e.displayText() = Connection refused, e.what() = Connection refused                                                 │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1273_171 because part a997169d040e813fc6f3049a156c1fbc_1009_1272_170 is not ready yet (log entry for that part is being processed). │                                                                                                                                                                  │
│ MetricsFree │                                                                                                                                                                                                 │                                                                                                                                                                  │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1277_175 because part a997169d040e813fc6f3049a156c1fbc_1009_1276_174 is not ready yet (log entry for that part is being processed). │                                                                                                                                                                  │
│ MetricsFree │ Not merging into part a997169d040e813fc6f3049a156c1fbc_1009_1279_177 because part a997169d040e813fc6f3049a156c1fbc_1009_1278_176 is not ready yet (log entry for that part is being processed). │ Code: 234, e.displayText() = DB::Exception: No active replica has part a997169d040e813fc6f3049a156c1fbc_1009_1279_177 or covering part, e.what() = DB::Exception │
│ MetricsFree │ Not merging into part 561fcdf2f7bd21b8eb12a475d96caf5d_552_579_5 because part 561fcdf2f7bd21b8eb12a475d96caf5d_552_573_4 is not ready yet (log entry for that part is being processed).         │ Code: 234, e.displayText() = DB::Exception: No active replica has part 561fcdf2f7bd21b8eb12a475d96caf5d_552_579_5 or covering part, e.what() = DB::Exception     │
└─────────────┴─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

63 rows in set. Elapsed: 0.004 sec. 

```

**Comment by iameugenejo** [comment]
I'm seeing the same issue with 18.5.1. Has anyone looked into the issue yet?

**Comment by simPod** [comment]
Me too but no idea how to resume without recreating stucked tables.

**Comment by nvartolomei** [comment]
@simPod detach/attach affected partition, suggested by @alesapin some time ago.

**Comment by simPod** [comment]
@nvartolomei could you point me to the suggestion? Currently I'm experiencing these exceptions 

```Code: 234, e.displayText() = DB::Exception: No active replica has part 202002_449608_449608_0 or covering part (version 20.1.2.4 (official build))```

I always go and recreate the table from scratch.

**Comment by nvartolomei** [comment]
In that case I guess part name is 202002.

```sql
ALTER TABLE table_name DETACH PARTITION '202002';
SELECT sleep(3);
ALTER TABLE table_name ATTACH PARTITION '202002';
```

you can try with and without quotes.

**Comment by simPod** [comment]
Thanks, I'll try when that happens again. Had to drop tables this time :)

**Comment by alexey-milovidov** [comment]
The original issue is fixed in #11637, #11639, #11795.
